aptos-move/framework/aptos-framework/sources/account.move

```diff
@@ -107,6 +107,10 @@ module aptos_framework::account {
     const EINVALID_ORIGINATING_ADDRESS: u64 = 13;
     /// The signer capability doesn't exist at the given address
     const ENO_SUCH_SIGNER_CAPABILITY: u64 = 14;
+    /// An attempt to create a resource account on a claimed account
+    const ERESOURCE_ACCCOUNT_EXISTS: u64 = 15;
+    /// An attempt to create a resource account on an account that has a committed transaction
```

Comment:
Can we clarify the following here:
```
/// First, with probability 1/2^256, an attacker can find a secret key whose corresponding public key has a derived address that collides with the resource account's address. This would allow an attacker to take over the resource account.
/// With probability 1/2^128, an attacker can find a public key `source`, a seed `seed` and another public key `pk` such that address returned by `create_resource_account(source, seed)` collides with the address derived for `pk`.
/// This would allow an attacker to create a resource account which they could surreptitiously take over at a later point.
/// Fortunately, both of these attacks happen with negligible probability.
```

Reply:
I can add a blob over the create_resource_account that states this!

Reply:
I believe so, yes. In the first attack, the attacker has to solve the following puzzle:

1. He sees a resource account address denoted by `rsrc_addr` which was derived from some input `(source, seed)`.
2. He needs to find a secret key `sk` such that (roughly speaking) hash `sk.to_public_key()` equals `rsrc_addr`.
3. This is the same as breaking a property of the hash function called _target collision resistance (TCR)_ (see [some old notes of mine here](https://github.com/alinush/6.857-lecture-notes/blob/master/l05-hash-1.md#hash-function-properties)
4. To break TCR on a hash function of length $\lambda$ bits requires $2^\lambda$ hash computations in expectation.

**Note:** TCR might not be the right name here. It might be _second preimage resistance_ or _weak collision resistance_. I do not have cycles to look into the subtleties of all of these (related) notions.